src/flint/types/fmpz_mod_mpoly.pyx

```diff
@@ -0,0 +1,1192 @@
+                    raise ZeroDivisionError("fmpz_mod_mpoly division by zero")
+                other = fmpz_mod_mpoly(other, self.ctx)
+                res = create_fmpz_mod_mpoly(self.ctx)
+                fmpz_mod_mpoly_div(res.val, (<fmpz_mod_mpoly>self).val, (<fmpz_mod_mpoly>other).val, res.ctx.val)
+                return res
+        return NotImplemented
+
+    def __rfloordiv__(self, other):
+        cdef fmpz_mod_mpoly res
+        if not self.ctx.is_prime():
+            raise DomainError("division with non-prime modulus is not supported")
+        elif not self:
+            raise ZeroDivisionError("fmpz_mod_mpoly division by zero")
+        other = any_as_fmpz(other)
+        if other is not NotImplemented:
+            other = fmpz_mod_mpoly(other, self.ctx)
+            res = create_fmpz_mod_mpoly(self.ctx)
+            fmpz_mod_mpoly_div(res.val, (<fmpz_mod_mpoly>other).val,  self.val, res.ctx.val)
+            return res
+        return NotImplemented
+
+    def __truediv__(self, other):
+        cdef:
+            fmpz_mod_mpoly res, div
+
+        if not self.ctx.is_prime():
+            raise DomainError("division with non-prime modulus is not supported")
+        elif typecheck(other, fmpz_mod_mpoly):
+            if not other:
+                raise ZeroDivisionError("fmpz_mod_mpoly division by zero")
+            elif self.ctx is not (<fmpz_mod_mpoly>other).ctx:
+                raise IncompatibleContextError(f"{self.ctx} is not {(<fmpz_mod_mpoly>other).ctx}")
+
+            res = create_fmpz_mod_mpoly(self.ctx)
+            if fmpz_mod_mpoly_divides(res.val, self.val, (<fmpz_mod_mpoly>other).val, self.ctx.val):
+                return res
+            else:
+                raise DomainError("fmpz_mod_mpoly division is not exact")
+        else:
+            o = any_as_fmpz(other)
+            if o is NotImplemented:
+                return NotImplemented
+            elif not o:
+                raise ZeroDivisionError("fmpz_mod_mpoly division by zero")
+            res = create_fmpz_mod_mpoly(self.ctx)
+            div = create_fmpz_mod_mpoly(self.ctx)
+            fmpz_mod_mpoly_set_fmpz(div.val, (<fmpz>o).val, self.ctx.val)
+            if fmpz_mod_mpoly_divides(res.val, self.val, div.val, self.ctx.val):
+                return res
+            else:
+                raise DomainError("fmpz_mod_mpoly division is not exact")
+
+    def __rtruediv__(self, other):
+        cdef fmpz_mod_mpoly res
+        if not self.ctx.is_prime():
+            raise DomainError("division with non-prime modulus is not supported")
+        elif not self:
+            raise ZeroDivisionError("fmpz_mod_mpoly division by zero")
+        o = any_as_fmpz(other)
+        if o is NotImplemented:
+            return NotImplemented
+        res = create_fmpz_mod_mpoly(self.ctx)
+        fmpz_mod_mpoly_set_fmpz(res.val, (<fmpz>o).val, self.ctx.val)
+        return res / self
+
+    def __mod__(self, other):
+        return divmod(self, other)[1]
```

Comment:
It is not quite right to return `divmod(...)` here because it means that we never return `NotImplemented` which in turn means that the other type cannot override this with `__rmod__`.